src/serac/physics/integrators/traction_integrator.hpp

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file hyperelastic_traction_integrator.hpp
+ *
+ * @brief Custom MFEM integrator for nonlinear finite deformation traction loads
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+namespace serac::mfem_ext {
+
+/**
+ * @brief Custom MFEM integrator for nonlinear finite deformation traction loads
+ */
+class TractionIntegrator : public mfem::NonlinearFormIntegrator {
+public:
+  /**
+   * @brief Construct a new Hyperelastic Traction Integrator object
+   *
+   * @param[in] f The traction load coefficient
+   * @param[in] compute_on_reference Flag for computing the residual in the stress-free configuration vs. the deformed
+   * configuration
+   */
+  explicit TractionIntegrator(mfem::VectorCoefficient& f, bool compute_on_reference = false)
+      : traction_(f), compute_on_reference_(compute_on_reference)
+  {
+  }
+
+  /**
+   * @brief Assemble the nonlinear residual on a boundary element at a current state
+   *
+   * @param[in] element_1 The first element attached to the face
+   * @param[in] element_2 The second element attached to the face
```

Comment:
Why are there two elements?

Reply:
This is an MFEM-ism because these face integrators are also used for DG methods. the second element is the optional other element attached to this face if it is calculated on the interior of the domain. It is MFEM convention that this element is null on the boundary.